src/app/containers/ConsentBanner/useConsentBanners/index.test.js

```diff
@@ -0,0 +1,289 @@
+  global.window.location = { ...location };
+});
+
+beforeEach(() => {
+  global.document.domain = 'www.bbc.com';
+  global.window.location.origin = 'https://www.bbc.com';
+  Cookies.set(PRIVACY_COOKIE, DEFAULT_PRIVACY_COOKIE);
+  Cookies.set(EXPLICIT_COOKIE, DEFAULT_EXPLICIT_COOKIE);
+  Cookies.set(POLICY_COOKIE, DEFAULT_POLICY_COOKIE);
+});
+
+describe('useConsentBanners', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('on initial mount', () => {
+    it('should return showPrivacyBanner=FALSE when PRIVACY_COOKIE is current policy value', () => {
+      Cookies.set(PRIVACY_COOKIE, DEFAULT_PRIVACY_COOKIE);
+      cookieSetterSpy.mockClear();
+
+      const { result } = renderHook(() => useConsentBanners());
+
+      expect(result.current.showPrivacyBanner).toBe(false);
+      expect(fetch).not.toHaveBeenCalled();
+    });
+
+    it('should return showPrivacyBanner=FALSE when PRIVACY_COOKIE is anythingelse', () => {
+      Cookies.set(PRIVACY_COOKIE, 'anythingelse');
+      cookieSetterSpy.mockClear();
+      const { result } = renderHook(() => useConsentBanners());
+
+      expect(result.current.showPrivacyBanner).toBe(false);
+      expect(fetch).not.toHaveBeenCalled();
+    });
+
+    it('sets PRIVACY_COOKIE and returns showPrivacyBanner=TRUE when PRIVACY_COOKIE is 0', () => {
+      Cookies.set(PRIVACY_COOKIE, '0');
+      cookieSetterSpy.mockClear();
+      const { result } = renderHook(() => useConsentBanners());
+
+      expect(cookieSetterSpy).toHaveBeenCalledWith(
+        PRIVACY_COOKIE,
+        DEFAULT_PRIVACY_COOKIE,
+        {
+          expires: 365,
+          domain: '.bbc.com',
+        },
+      );
+      expect(result.current.showPrivacyBanner).toBe(true);
+      expect(fetch).not.toHaveBeenCalled();
+    });
+
+    it('sets PRIVACY_COOKIE and returns showPrivacyBanner=TRUE when PRIVACY_COOKIE is 1', () => {
+      Cookies.set(PRIVACY_COOKIE, '1');
+      cookieSetterSpy.mockClear();
+      const { result } = renderHook(() => useConsentBanners());
+
+      expect(cookieSetterSpy).toHaveBeenCalledWith(
+        PRIVACY_COOKIE,
+        DEFAULT_PRIVACY_COOKIE,
+        {
+          expires: 365,
+          domain: '.bbc.com',
+        },
+      );
+      expect(result.current.showPrivacyBanner).toBe(true);
+      expect(fetch).not.toHaveBeenCalled();
+    });
+
+    it('sets PRIVACY_COOKIE and returns showPrivacyBanner=TRUE when cookie is null', () => {
+      Cookies.set(PRIVACY_COOKIE, null);
+      cookieSetterSpy.mockClear();
+      const { result } = renderHook(() => useConsentBanners());
+
+      expect(cookieSetterSpy).toHaveBeenCalledWith(
+        PRIVACY_COOKIE,
+        DEFAULT_PRIVACY_COOKIE,
+        {
+          expires: 365,
+          domain: '.bbc.com',
+        },
+      );
+      expect(result.current.showPrivacyBanner).toBe(true);
+      expect(fetch).not.toHaveBeenCalled();
+    });
+
+    it('sets PRIVACY_COOKIE without domain restrictions', () => {
```

Comment:
What is this test covering? Not sure what 'domain restricitons' are?

Reply:
<img width="1331" alt="Screenshot 2021-08-20 at 09 09 51" src="https://user-images.githubusercontent.com/4798332/130202125-ae381ed5-f366-440b-91ff-3229abd9d830.png">

Judging from the previous test I think it's just testing that the domain is set in the cookie.